core/src/main/java/org/infinispan/persistence/async/AsyncCacheWriter.java

```diff
@@ -151,18 +151,22 @@ public boolean delete(Object key) {
    }
 
    protected void applyModificationsSync(List<Modification> mods) throws PersistenceException {
+      List<MarshalledEntry> entriesToWrite = new ArrayList<>();
```

Comment:
I believe this already happens when the modification is created and then stored in the `modifications` map based upon the entries key.